### ISSUE TITLE
Build Warning - "cast from pointer to integer of different size [-Wint-to-pointer-cast]"

### ISSUE THREAD

**Issue by Brahmasmi** [projectNe10/Ne10#123]
Building Ne10 for Android using NDK version r10e following BuildingNe10.txt instructions with the following options

```
ANDROID_API_LEVEL=21
ARM_ANDROID_TOOLCHAIN_VERSION=4.9
NE10_ANDROID_TARGET_ARCH=aarch64
```

outputs the following warning

```
...
[ 14%] Building C object modules/CMakeFiles/NE10.dir/dsp/NE10_fir.c.o
Ne10/modules/dsp/NE10_fir.c: In function 'ne10_circular_read_float':
Ne10/modules/dsp/NE10_fir.c:1333:15: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
     dst_end = (ne10_int32_t) (dst_base + dst_length);
               ^
Ne10/modules/dsp/NE10_fir.c:1346:20: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
         if (dst == (ne10_int32_t *) dst_end)
                    ^
...
```

I was wondering whether this warning warrants a resolution.

**Comment by Phillip-Wang** [comment]
Hi @Brahmasmi , it seems we have mixed int32 and pointer here.

**Comment by dramosz** [comment]
I am also wondering whether this warning warrants a resolution?

**Comment by Brahmasmi** [comment]
Hi @Phillip-Wang ,

Would it be possible to get an approximate idea on the timeline for resolution of this warning?

Thanks.

**Comment by joesavage** [comment]
Perhaps I'm missing something here, but there doesn't seem to be a valid reason for `dst_end` to be of type `ne10_int32_t` at all here. It would be far more sensible for it to be of the same pointer type as `dst`, i.e. an`ne10_int32_t *`, which would remove the unnecessary casts and these annoying warnings.

There are plenty of other warnings produced by compiling the codebase with `-Wall` too that should probably be fixed up at some point. Unused variables, type mis-matches, strict aliasing issues, you name it.

**Comment by wookey** [comment]
Cheers for all those warning fixes Joe. I have included them in the debian package.

**Comment by Brahmasmi** [comment]
Would removing the -Wall flag and adding the -fno-* flags cause any problems or is this a stopgap measure till the warnings are fixed? I ask this because from what I remember -Wall allows all warnings to be flagged (and subsequently fixed) and that -fno-* usually suppresses some warnings.

**Comment by joesavage** [comment]
I'm not sure I completely understand your question — are you referring to my patch, or to something else? From the perspective of the codebase as a whole, removing the `-Wall` flag would very much be sweeping the underlying problems under the rug.

In my patch, I've fixed all the warnings I was seeing in my local compilation environment, and this included adding the `-fno-strict-aliasing` flag — not to suppress any warnings per se, but to *resolve* some warnings by preventing strict aliasing optimisations which might cause some of the code to run incorrectly. This also included adding `-Wno-unused-but-set-variable` to suppress one particular type of warning, but only because I think these warnings are mostly not worth fixing (in some places, the code is better with the things that the warning complains about).

If you're looking for a "quick fix" to the warnings in your local development environment, suppressing them (ensuring `-Wall` is off, adding some `-Wno-*` flags) and adding `-fno-strict-aliasing` is probably the fastest way to go about it.

**Comment by joesavage** [comment]
Fixed in 94677e6c0b761c64cabd7ac697ba9c6b350a9902
